src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -417,8 +416,10 @@ private static boolean shouldBeAloneOnLineWithNotAloneOption(Details details,
     private static boolean isAloneOnLine(Details details, String targetSrcLine) {
         final DetailAST rcurly = details.rcurly;
         final DetailAST nextToken = details.nextToken;
-        return (!TokenUtil.areOnSameLine(rcurly, nextToken) || skipDoubleBraceInstInit(details))
-                && CommonUtil.hasWhitespaceBefore(details.rcurly.getColumnNo(), targetSrcLine);
+        return (nextToken == null || !TokenUtil.areOnSameLine(rcurly,
+                nextToken) || skipDoubleBraceInstInit(details))
```

Comment:
`nextToken)`
please move it to previous line, now it looks a bit ugly

Reply:
Done in #9324